snippets/src/offerCreate-v2.ts

```diff
@@ -0,0 +1,16 @@
+const client = new WsClient({server: `wss://xrpl.ws`})
+const wallet: Wallet = Wallet.walletFromSeed("sEd…")
+
+const tx: OfferCreateTransaction = {
+    account: "r…",
+    taker_gets: "XRP",
+    taker_pays: {...}
+}
+
+const preparedTransaction = client.autofill(tx)
```

Comment:
I don't think that `autofill` needs to be front-and-center anymore. In my mind, `autofill` becomes a public method that is effectively never called, and would be renamed something like `autofillUnsignedTransaction`. 

My rationale is that there are two flows, generally, as far as I can see:

1) I just want to autofill this transaction and get it out the door. For that use case, we would have a method on the client called `submitTransaction`, which takes a transaction and a wallet. This will a) autofill, b) sign, c) submit. This will be the vast majority of use cases.
2) I have specific needs involving signing. In this case, we provide a module of signer helpers and another method on client called `submitSignedTransaction` which expects a user has already dealt with signing.

As a result, I think that explicit use of `autofill` is really unnecessary. We can reduce the number of method calls by one for each transaction this way!

Reply:
It's an interesting idea. (1) is common for tutorials and examples, and I'm in favor of a convenience method for cases where  that's what the developer wants. (2) is a lot easier when you have autofill (or [prepareTransaction](https://github.com/ripple/ripple-lib/blob/develop/src/transaction/utils.ts#L110)). By the way, there is no concept or method called `autofill` in ripple-lib; it is just one of the things handled by the `prepare*` methods. I would prefer to keep the current name (prepareTransaction) unless we have a really good reason to change it.

Unfortunately(?), I think there are a lot of use cases where signing benefits from being a separate step. In production systems, signing is a risky and sensitive operation:
- For security purposes, we want developers to know when and where signing is taking place, and we want users to know exactly what they are signing.
- Manual signing flows typically involve air-gapped machines and "offline" methods of transferring unsigned and signed transactions (flash drives, QR codes, Bluetooth).
- Automated signing flows are sometimes separate from the rest of the application logic. While almost everything else can be done in parallel and/or scaled across multiple application servers, signing often needs to be done synchronously on a secure system that keeps track of what was signed and when.
- Sometimes the signing machine is isolated from the wider internet and only communicates with the app server.

Reply:
Also, there are flows where ripple-lib is used for everything except signing, and signing is handled by a separate device. XUMM, Ledger hardware wallets, Trezor, ...

Reply:
For those usecases, the `autofill` method will still be a publicly exposed method that devs can use, as @ledhed2222 mentioned, but it's unnecessary to use the method in tutorials and the like, because any devs that want that level of control will be willing to dig through the docs a bit to figure out how to do it.

Reply:
As mentioned, "autofill" isn't really a thing in ripple-lib. I guess "autofill" comes from xrpl-py.

In ripple-lib, I think prepareTransaction handles the equivalent step. In flows that don't use prepareTransaction, how should users specify maxLedgerVersionOffset and signersCount?

Reply:
In the current proposed flow, `autofill` would basically replace `prepareTransaction`, and could be used similarly.